app/src/main/java/edu/drexel/lapcounter/lapcounter/frontend/LoadingActivity.java

```diff
@@ -29,4 +48,52 @@ public void finishLoading(View view) {
         Intent intent = new Intent(this, WelcomeActivity.class);
         startActivity(intent);
     }
+
+    private void initDatabase() {
+        // Initialize the enum tables. ====================================
+        try
+        {
+            WorkoutRepository workoutRepo = new WorkoutRepository(getApplication());
+            workoutRepo.initUnitsTable();
+
+
+            TransitionRepository transRepo = new TransitionRepository(getApplication());
+            transRepo.initStatesTable();
+        }
+        catch(Exception e)
+        {
+            //Not sure what to do here
```

Comment:
Depends on the exception.
We know that the `init*Table` methods can throw `ExecutionException`s or `InterruptedException`s in addition to some truly oddball unexpected exceptions.

We need a better sense of when and why those methods would throw these exceptions, so we should be error logging the string messages of the thrown exceptions.